### DISCUSSION TITLE
Please communicate with me through this GitHub discussion

### DISCUSSION THREAD

**Discussion by anlhu** [casa-business-manager/CASA#52] [General]
It is the only way I can message

**Comment by anlhu** [comment]
Here is my scrum update for Tuesday

I have made substantial progress on the roles story. The frontend task is finished and I started on integration, but it turns out there is more to the frontend that I didn't consider so I'm working longer than expected.

Before Thursday I should be done with the roles integration for sure. After this task I'm not sure what to work on, but I think I can check in again on Thursday. 

There's still the issue of what permissions to add and how to enforce them. Also, I'm not sure how we want to handle sub roles when we delete a role. Should we just make all the sub roles into sub roles of the parent role? @IanChong1 Please let me know what you think about these. Then another blocker is sometimes I can't connect to GitHub to push so it's a bit intermittent. 

@catdiehl Could you please send me everyone else's updates? I'm totally in the dark now cuz I can't access discord.

**Reply by anlhu** [reply]
Also here is a demo of what I have on `roles-integration`:
![image](https://github.com/user-attachments/assets/e6a9d114-a508-42f6-a1a4-ed3b581c1ce8)
![image](https://github.com/user-attachments/assets/45dd776b-5a46-4c54-be83-a02e40109d29)
![image](https://github.com/user-attachments/assets/3399550b-09ac-4b67-8356-337f5a4d6e2e)
![image](https://github.com/user-attachments/assets/7d7a299d-e24e-4285-9164-76b6e8f897b3)

For the last picture, I plan to use the right side of the screen to bring up a new edit menu, and then you press "create"/"save" to confirm with the backend. 

For the coloring, I'm planning to make it so the roles that the user has will be highlighted, the subtrees under them will be like a dark gray, and then all the roles above them will be out of focus/light gray.

**Reply by catdiehl** [reply]
hey Andrew I saw in the PR that you have separating APIUtils.js into functions by controller, would you like me to do that?

**Reply by catdiehl** [reply]
just let me know what you think I should name the new .js for each controller

**Reply by anlhu** [reply]
Yeah sure, it shouldnt be much work if your VS Code automatically updates imports for you.

i think we can name them like "EventAPI.js" and "OrganizationAPI.js" etc..
